src/discord/discord.service.ts

```diff
@@ -1,26 +1,73 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { CreateDiscordDto } from './dto/create-discord.dto';
+import { ReadDiscordDto } from './dto/read-discord.dto';
 import { UpdateDiscordDto } from './dto/update-discord.dto';
 
 @Injectable()
 export class DiscordService {
+  private readonly discord: ReadDiscordDto[] = [];
   create(createDiscordDto: CreateDiscordDto) {
-    return 'This action adds a new discord';
+    const discordUser = {
+      id: Date.now(),
+      username: createDiscordDto.username,
+      bio: { ...createDiscordDto.bio },
+      createdOn: new Date(Date.now()),
+      updatedOn: new Date(Date.now()),
+    };
+    this.discord.push(discordUser);
+    return 'User added successfully!';
   }
 
   findAll() {
-    return `This action returns all discord`;
+    return [...this.discord];
   }
 
   findOne(id: number) {
-    return `This action returns a #${id} discord`;
+    const discordUser = this.findDiscord(id);
```

Comment:
I think this private function call could be replaced with a javascript `find`

Reply:
In the future, the DB will be doing it, so I think it might be optimisation too soon

Reply:
> I think this private function call could be replaced with a javascript `find`

It's also doing error handling 
```ts
if (!discordUser) {
      throw new NotFoundException('User Not Found');
    }
```
Which I'll have to repeat on multiple locations.  How about I keep the function and replace `findIndex` with `find`? (Although it won't improve performance even when we have large data)

Reply:
the `find` is temporary, because it will be handled by the DB. We can leave this function for now if you like